bento-platform/bento
# Feature/cbioportal

## New Features:
- Add cBioPortal Containers to BentoV2
- Add a cache directory to the WES container for storing some genomic data necessary to the vcf2maf conversion
- Add mounted volume from the DRS data directory to the WES container for more efficient file processing when vcf files are stored in DRS
- Configurable ports added to some containers in DEV mode and fix conflict between those (e.g. Katsu DB and DRS)
- New scripts added to make file: populate WES cache directory, download data for cBioPortal seed database

## Changes to .env:
- new variables:
  - CBIOPORTAL_DOMAIN
  - CBIOPORTAL_PUBLIC_URL
  - CBIOPORTAL_PUBLIC_URL_TRAILING
  - BENTOV2_USE_CBIOPORTAL
  - BENTOV2_KATSU_DB_INTERNAL_PORT
  - BENTOV2_KATSU_DB_EXTERNAL_PORT
  - BENTOV2_WES_VOL_CACHE_DIR
  - BENTOV2_WES_VEP_ENSEMBL_RELEASE_VERSION
  - CBIOPORTAL_* variables from the end of the environment file

## How to test:
Update the branches for the following projects (in .env, remove the xxx_TAG value and set only the xxx_BRANCH value for the following):
- `KATSU_BRANCH=features/cbio_export_workflow`
- `BENTO_WES_BRANCH=features/cbioportal` IMPORTANT: this container MUST be rebuilt. This takes about 10min (for the installation of ENSEMBL_VEP)
- `BENTO_WEB_BRANCH=cbio_export`

The following scripts must be tested.
- `make data-dirs` adds a cache directory to data/wes/, and cBioPortal related dirs in data/cbioportal/(data, study,database, session_database)
- `make init-cbioportal` downloads the seed databases to data/cbioportal/data (cgds.sql and seed.sql.gz)
- `make init-wes-cache` creates the wes cache. It requires the `features/cbioportal` branch of wes to work. It takes 3h to complete. The result can be seen in the data/wes/cache directory
- `make run-cbioportal` launches the 4 cBioPortal containers. They should be accessible behind https://cbioportal.bentov2.local/. I am having trouble with the gateway which redirects me to keycloak instead. I can access locally at http://localhost:8089 instead.

To test the integration with Bento:

1. Generate maf files: In Admin>Data Manager, select the tab [Analysis], select the dataset and run the vcf2maf conversion. This workflow can take several minutes to run (about 3min per vcf file) depending on the number of vcf files to process. It is somehow possible to spy on the progress of the workflow by looking at the content of the folder in ./data/wes/tmp/ID_OF_THE_RUN/
![Screen Shot 2022-09-30 at 1 53 04 PM](https://user-images.githubusercontent.com/3930632/193331969-702a496e-0dd0-4b01-b936-4f6660b3c727.png)

2. Generate the data to import into cBioPortal
In the data manager, select the tabl [Export], select the dataset and run the export to cBioPortal script. The process can be spied using the same strategy as previously
3. Finally the data can be imported into cBioPortal by manually copying the content of the former folder to ./data/cbioportal/study and running the following command `docker-compose run cbioportal metaImport.py -u http://cbioportal:8080/ -s study/cf46bd61-6d95-4248-b3c5-fde31105ca07/ -o` where `cf46bd61-6d95-4248-b3c5-fde31105ca07` is the id of the dataset that was exported (this manual step should be made simpler in the future)
